aiohttp/web.py

```diff
@@ -263,6 +263,24 @@ def __repr__(self):
         return "<Application 0x{:x}>".format(id(self))
 
 
+############################################################
+# Helper functions
+############################################################
+
+def named_url(app_or_request, name, **kwargs):
+    app = getattr(app_or_request, 'app', app_or_request)
```

Comment:
`isinstance` check is safer than duck typing.
Parameter which is neither application nor request should raise `TypeError`